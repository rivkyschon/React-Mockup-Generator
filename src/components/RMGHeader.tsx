// src/components/RMGHeader.tsx
import React from 'react';

interface RMGHeaderProps {
  text: string;
}

const RMGHeader: React.FC<RMGHeaderProps> = ({ text }) => {
  return (
    <h1 className="font-bold text-3xl text-gray-900">{text}</h1>
  );
};

export default RMGHeader;
