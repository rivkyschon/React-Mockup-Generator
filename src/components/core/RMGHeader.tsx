// src/components/RMGHeader.tsx
import React from 'react';

interface RMGHeaderProps {
  text: string;
  className?: string;
}

const RMGHeader: React.FC<RMGHeaderProps> = ({text, className}) => {
  return (
    <h1
      className={
        className || 'text-center text-4xl font-bold text-gray-800 mb-6'
      }>
      {text}
    </h1>
  );
};

export default RMGHeader;
