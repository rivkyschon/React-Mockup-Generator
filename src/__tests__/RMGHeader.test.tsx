// src/components/__tests__/RMGHeader.test.tsx

import React from 'react';
import {render} from '@testing-library/react';
import RMGHeader from '../components/RMGHeader';

describe('RMGHeader', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<RMGHeader text="Test Header" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    const {getByText} = render(<RMGHeader text="Test Header" />);
    // expect(getByText('Test Header')).toBeInTheDocument();
  });
});
