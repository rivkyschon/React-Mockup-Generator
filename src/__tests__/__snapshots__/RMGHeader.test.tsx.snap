// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGHeader renders correctly 1`] = `
<DocumentFragment>
  <h1
    class=" text-center font-bold text-4xl text-gray-900 mb-6"
  >
    Test Header
  </h1>
</DocumentFragment>
`;
